From 8b0fb0222efd04dc54801421643c426effcee2ee Mon Sep 17 00:00:00 2001
From: Sebastiano Barezzi <seba@sebaubuntu.dev>
Date: Fri, 12 Jul 2024 03:10:44 +0200
Subject: [PATCH] kernel: Better x86 support

- Mention the right kernel image on documentation
- Build dtbs only when the dts folder exists, Google enables CONFIG_OF
  on x86_64 GKI config, this is also the same check done on Linux
  Makefile

Change-Id: I1b0f7a39b0b00f2a4516d2cff79e591136ee3af8
(cherry picked from commit 3fe68ad58a11a183bc5113b9535f27048160f18c)
---
 build/tasks/kernel.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
index bc907679f0..2fd6ec5fb0 100644
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -34,6 +34,7 @@
 #   BOARD_KERNEL_IMAGE_NAME            = Built image name
 #                                          for ARM use: zImage
 #                                          for ARM64 use: Image.gz
+#                                          for x86 use: bzImage
 #                                          for uncompressed use: Image
 #                                          If using an appended DT, append '-dtb'
 #                                          to the end of the image name.
@@ -448,7 +449,7 @@ $(KERNEL_CONFIG): $(KERNEL_OUT) $(ALL_KERNEL_DEFCONFIG_SRCS)
 $(TARGET_PREBUILT_INT_KERNEL): $(KERNEL_CONFIG) $(DEPMOD) $(DTC)
 	@echo "Building Kernel Image ($(BOARD_KERNEL_IMAGE_NAME))"
 	$(call make-kernel-target,$(BOARD_KERNEL_IMAGE_NAME))
-	$(hide) if grep -q '^CONFIG_OF=y' $(KERNEL_CONFIG); then \
+	$(hide) if [ -d "$(KERNEL_SRC)/arch/$(KERNEL_ARCH)/boot/dts/" ]; then \
 			echo "Building DTBs"; \
 			$(call make-kernel-target,dtbs); \
 		fi
